spark/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -454,15 +459,32 @@ private static UnsafeProjection projection(Schema finalSchema, Schema readSchema
     }
 
     private CloseableIterable<InternalRow> newParquetIterable(InputFile location,
-                                                            FileScanTask task,
-                                                            Schema readSchema) {
+                                                              FileScanTask task,
+                                                              Schema readSchema) {
       return Parquet.read(location)
           .project(readSchema)
           .split(task.start(), task.length())
           .createReaderFunc(fileSchema -> SparkParquetReaders.buildReader(readSchema, fileSchema))
           .filter(task.residual())
           .build();
     }
+
+    private CloseableIterable<InternalRow> newOrcIterable(InputFile location,
```

Comment:
I think this class should be defined with the reader, not here in the Spark reader. The ORC support should provide a CloseableIterable, not just an iterator.

Reply:
I agree. Let me re-work this one.

Reply:
@edgarRd, Looks like this one still needs to be done.

Reply:
I've refactored this code. I think now it should follow a similar pattern to the other supported formats. I've additionally removed write support for now.